lib/auth/trustedcluster.go

```diff
@@ -322,6 +322,23 @@ func (a *Server) addCertAuthorities(trustedCluster types.TrustedCluster, remoteC
 	return nil
 }
 
+func (a *Server) updateUserCARoleMap(ctx context.Context, existingCluster, trustedCluster types.TrustedCluster) error {
+	certAuthority, err := a.GetCertAuthority(ctx, types.CertAuthID{
+		Type:       types.UserCA,
+		DomainName: existingCluster.GetName(),
+	}, false)
+	if err != nil {
+		return trace.Wrap(err)
+	}
+	if certAuthority.GetType() == types.UserCA {
```

Comment:
Why is this check needed? Aren't you querying specifically `types.UserCA` above?